docs/source/error_code_list.rst

```diff
@@ -1149,6 +1149,26 @@ types you expect.
 
 See :ref:`overloading <function-overloading>` for more explanation.
 
+
+.. _code-overloaded-function-matching:
+
+Check error code of overload function signature match
```

Comment:
The documentation should say something about when this error happens and what to do about it.